applications/vanilla/controllers/api/CommentsApiController.php

```diff
@@ -184,6 +184,57 @@ public function get($id, array $query) {
         return $result;
     }
 
+    /**
+     * Get a comments quote data.
+     *
+     * @param int $id The ID of the comment.
+     *
+     * @return array The comment quote data.
+     *
+     * @throws NotFoundException If the record with the given ID can't be found.
+     * @throws \Exception if no session is available.
+     * @throws \Vanilla\Exception\PermissionException if the user does not have the specified permission(s).
+     * @throws \Garden\Schema\ValidationException If the output schema is configured incorrectly.
+     */
+    public function get_quote($id) {
+        $this->permission();
+
+        $this->idParamSchema();
+        $in = $this->schema([], ['in'])->setDescription('Get a comments embed data.');
+        $out = $this->schema($this->quoteSchema(), 'out');
+
+        $comment = $this->commentByID($id);
+        if ($comment['InsertUserID'] !== $this->getSession()->UserID) {
+            $discussion = $this->discussionByID($comment['DiscussionID']);
+            $this->discussionModel->categoryPermission('Vanilla.Discussions.View', $discussion['CategoryID']);
+        }
+
+        $comment['Url'] = commentUrl($comment);
+        $isRich = $comment['Format'] === 'Rich';
+        $comment['bodyRaw'] = $isRich ? json_decode($comment['Body'], true) : $comment['Body'];
+
+        $this->userModel->expandUsers($comment, ['InsertUserID'], ['expand' => true]);
+        $result = $out->validate($comment);
+        return $result;
+    }
+
+    /**
+     * Get the schema for comment quote data.
+     *
+     * @return Schema
+     */
+    private function quoteSchema(): Schema {
+        return Schema::parse([
+            'commentID:i' => 'The ID of the comment.',
+            'bodyRaw:s|a' => 'The raw body of the comment. This can be an array of rich operations or a string for other formats',
```

Comment:
Vanilla is trying to improve its OpenAPI support. Unfortunately, OpenAPI doesn't support multiple types for parameters. If there's any way this can work with just a string, that'd be the way to go.